vllm/executor/multiproc_worker_utils.py

```diff
@@ -0,0 +1,265 @@
+import asyncio
+import multiprocessing
+import os
+import sys
+import threading
+import traceback
+import uuid
+from dataclasses import dataclass
+from multiprocessing import Queue
+from multiprocessing.connection import wait
+from multiprocessing.process import BaseProcess
+from typing import (Any, Callable, Dict, Generic, List, Optional, TextIO,
+                    TypeVar, Union)
+
+from vllm.logger import init_logger
+from vllm.utils import enable_trace_function_call_for_thread
+
+logger = init_logger(__name__)
+
+T = TypeVar('T')
+
+_TERMINATE = "TERMINATE"  # sentinel
+
+# ANSI color codes
+CYAN = '\033[1;36m'
+RESET = '\033[0;0m'
+
+# Use dedicated multiprocess context for workers.
+# Both spawn and fork work
+mp_method = os.getenv("VLLM_WORKER_MULTIPROC_METHOD", "fork")
```

Comment:
@zhuohan123 re your [prior comment](https://github.com/vllm-project/vllm/pull/3466#discussion_r1575573260), `fork` is actually about twice as fast as `spawn`, so I thought maybe best to keep this as the default (there are scenarios where start-up time is important and the difference is at least 1-2 seconds).

Reply:
Hmm I feel like using spawn default is better (and we can suggest to change fork if necessary). 

Afaik (including my experience from other project), fork can cause many random issues, espeically when it is used with cuda (like random hang kind of thing). Feel like defaulting stable option is better?